services/nominal-osm.js

```diff
@@ -0,0 +1,114 @@
+import axios from 'axios';
+
+const BASE_URL = 'https://nominatim.openstreetmap.org/search';
+
+const https = require('https');
+
+let cacheAgent;
+
+function getAgent() {
+  if (!cacheAgent) {
+    cacheAgent = new https.Agent({ keepAlive: true });
+  }
+
+  return cacheAgent;
+}
+
+function emptyLocation() {
+  return {
+    type: 'Point',
+    coordinates: { longitude: undefined, latitude: undefined },
+  };
+}
+
+async function fetchGeocoordinateFromBrazilLocation({
+  state,
+  city,
+  street,
+  cep,
+  neighborhood,
+}) {
+  const agent = getAgent();
+  const country = 'Brasil';
+  let streetUse = '';
+
+  if (street) {
+    if (street.includes('-')) {
+      const streetSplit = street.split('-');
+      if (streetSplit && streetSplit[0]) {
+        const [streetRes] = streetSplit;
+        streetUse = streetRes;
+      }
+    } else {
+      streetUse = street;
+    }
+  }
+
+  const headers = {
+    'User-Agent': 'brasil-api-cep-v2',
+  };
+
+  try {
+    const params = {
+      country,
+      city,
+      state,
+      treet: streetUse,
```

Comment:
Hi @enieber, I think there's a little typo here (`treet` instead of `street`)

Reply:
up